### ISSUE TITLE
Solutions for HW 6, 7 missing - causing Travis to fail

### ISSUE THREAD

**Issue by ethanmad** [c4cs/c4cs.github.io#235]
The code block at https://github.com/c4cs/c4cs.github.io/blob/d60f6d49c67d1979f6dd0262778a1ff71fd1f2bd/_layouts/syllabus.html#L108:

```html
 {% if now > solutionRelease %}
        <hr />
        <p><a href="static/f16/solutions/c4cs-wk{{ page.week }}-solutions.pdf">Solutions&nbsp;{{ page.week }} [pdf]</a></p>
      {% endif %}
```

is causing Jekyll to expect there to exist PDFs for the solutions for HW 6, 7 since the solution releases defined in the syllabus files ([6](https://github.com/c4cs/c4cs.github.io/blob/master/_syllabus/f16/week06.md), [7](https://github.com/c4cs/c4cs.github.io/blob/master/_syllabus/f16/week07.md)) are now before the present.

Since these files do not exist, the Travis build fails: `internally linking to static/f16/solutions/c4cs-wk6-solutions.pdf, which does not exist (line 1070)` and `internally linking to static/f16/solutions/c4cs-wk7-solutions.pdf, which does not exist (line 1204)`.

A solution would be checking in the linked code block that the file does exist, which seems more complicated (and inefficient, though that doesn't matter because it runs at comile-time) than it should:
http://stackoverflow.com/questions/16528783/check-for-existence-of-file-using-jekyll

**Comment by ppannuto** [comment]
Hmm.. this one goes both ways. In this case, the Travis build failing is probably a good thing. It's the reminder to me to post solutions when I forget!

In this case

 - I forgot to push after committing for week 6 :blush: [just fixed]
 - I need to prepare them for week 7 (!) [fixing..]

That said, it would be a nice code idiom to be able to drop for some other things.

**Comment by ethanmad** [comment]
(The following is with all respect to you and the staff--these are just some brief follow-up thoughts on the issue.)

Thinking about this again, it raises the question: why do the time check at all?

If you publish solutions in this repository before that time, then they're available, even if they're not on the website.

Also, since the check happens at compile-time (which, for GitHub Pages, occurs upon receiving a push, I think), the website won't show the solutions until the _next_ rebuilding of the site after the time has passed if you push the solutions in advance. If you're consistently posting them after the specified time, then the check is actually not doing anything.

I might be missing something, but other than serving as a reminder to post solutions (and there are probably nicer ways than having Travis email you after every commit), it seems to me that the time check is unnecessary complexity.

**Comment by ppannuto** [comment]
(Totally happy to discuss stuff like this, always looking to improve)

My philosophy leans pretty heavily towards the honor code and the principles it espouses. Yes, link-guessing would work to get at solutions posted in advance of the timestamp "releasing" the solutions. Or looking through the repository. Or looking up the course archives for previous semesters, most homeworks didn't change too much. The point expressed by the "release" mechanism is that there's a time you're expected to look at the solutions. Their purpose is to help explain a homework that you may not have understood. Ultimately, the point of the homework exercises are for people to learn. There are numerous ways for people to cheat and I'm not really interested in policing them.

Outside of philosophy, the build check does do what it's expected to do because the links are _internal_ links (i.e. they link to `/static/f16/solutions/c4cs...pdf`, not `https://c4cs.github.io/static/...`), which means when the site is compiled as part of the Travis build, the expected link targets are present. [example: when I uploaded solutions the build went green https://travis-ci.org/c4cs/c4cs.github.io/builds/172487092]

Now, is the "release time" thing unnecessary complexity? I'd say absolutely :D. I already had time-release code lying around from the update mechanism, though, so it was very little work to add. And I think there's something elegant about all of the homeworks releasing themselves automatically.

Checking for broken links has caught typos and other silly errors more often than I care to admit, and I wrote none of that code, just grabbed something from online, which is pretty much the dream.

**Comment by ethanmad** [comment]
> There are numerous ways for people to cheat and I'm not really interested in policing them.

Makes sense, and I'm a fan of your philosophy as someone here to learn.

> the build check does do what it's expected to do because the links are internal links 

What I meant was the `{% if now > solutionRelease %}` check evaluates at build time, so the solutions won't magically appear on the course site the moment the statement evaluates to true, but instead when the site is next rebuilt. But since the site is being rebuilt multiple times almost every day, this is not really an important point and the release mechanism is just plain cool.

**Comment by ethanmad** [comment]
Closed by 1259a46d70c328851281db03e51dc1b7f40797c4.
